### PR TITLE
Add - Remove List

### PR DESCRIPTION
**Pull request by Itswali**
In This PR:-
* Removed all hardcoded items from the tasks array.
* Create a new JavaScript file for new functionality.
* Implemented a function for adding a new task.
* Implemented a function for deleting a task.
* Deleting a task updates all remaining items' indexes, so they represent the current list order and are unique
* All data stored and used from Local storage.

### CODE REVIEW

File: src/modules/task.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export default class Task {
+  constructor(id, description, completed) {
+    this.id = id;
+    this.description = description;
+    this.completed = completed;
+  }
```
Comment (mckent05) [inline src/modules/task.js]: - According to the project requirements, all task should have `index`, `completed` and `description` properties, please change the `id` property to `index` 🙏 🙏

File: src/index.js [hunk 2]
```diff
@@ -1,46 +1,157 @@
...
+    const currentList = store.getList();
+    const filteredList = currentList.filter((task) => task.id !== id);
+    localStorage.setItem('list', JSON.stringify(filteredList));
+    store.resetIds();
+    const tasks = getUItasks();
+    tasks.forEach((task) => task.remove());
+
+    UI.displayList();
+  }
+}
```
Comment (mckent05) [inline src/index.js:141]: - I get an error when I try removing/deleting a task, please reference the picture I attached below, also please remember to update the task index and save all changes made to local storage. Please fix this 🙏 🙏 

Your project:
ewgeg

File: src/index.js [hunk 3]
```diff
@@ -1,46 +1,157 @@
...
+  const id = parseInt(idstring, 10);
+  const list = store.getList();
+  for (let i = 0; i < list.length; i += 1) {
+    const task = list[i];
+    if (task.id === id) {
+      list[i].description = text;
+    }
+  }
+  localStorage.setItem('list', JSON.stringify(list));
+}
```
Comment (mckent05) [inline src/index.js]: - When I try updating a task, it doesn't get updated in the local storage, please fix this issue 🙏 🙏 

Your project:
wdw

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Itswali,

# STATUS: CHANGES REQUIRED ♻️ ♻️  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️  
- PR title and summary are descriptive 👏 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @Itswali👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**